### ISSUE TITLE
Enhance Database Schema for Users model

### ISSUE THREAD

**Issue by jealapera** [jealapera/todo-app-api#4]
Objective:
- Enhance the Users model schema to improve functionality, security, and scalability.

Requirements:
- Email Verification Status:
   - Implement a field to indicate whether the user's email has been verified.
- Verification Code:
   - Introduce a field to store the verification code sent to the user's email during the verification process.
- Profile:
   - Introduce a nested object field named "profile" with keys:
      - Photo: Store the user's profile photo.
      - Full Name: Store the user's full name.
      - Bio: Include a field for the user's brief biography or description.
      - Account Creation Date:
   - Implement a field to record the date and time when the user's account was created.
- Username:
   - Introduce a field for the user's username.
- Last Login Date:
   - Track the date and time of the user's last login. Additionally, archive a historical record of login dates for up to 7 days.
- Two-Factor Authentication (2FA):
   - Introduce a field to enable or disable two-factor authentication.
- Password Reset Token:
   - Implement a field to securely store a token for password reset.
- Account Status:
   - Include a field to indicate the status of the user's account.
- Timezone:
   - Introduce a field to capture the user's timezone preferences.
- Language Preference:
   - Include a field to store the user's preferred language.

Implementation:
1. Schema Update: Enhance the Users model schema within the `userModel.js` file located in the `/models` folder using Mongoose, incorporating the mentioned requirements.
2. Validation:  Implement validation checks within the model schema to enforce business rules for the new fields.
3. API Integration: Update API endpoints and controllers to support CRUD operations with the enhanced model schema.
4. Testing: Conduct comprehensive testing to validate new schema functionality, including edge cases and error handling.

Documentation:
- Update the API documentation for the User model to incorporate updates regarding schema modifications and the utilization of newly added fields.

**Comment by iamcubs** [comment]
Hi @jealapera cc @kimmayorga ,  in regards with this part of the task

Migration File Creation:
Enhance the Users model schema by creating a migration file.
This migration file should reflect the changes needed to accommodate the new requirements.
Refer to the [Knex.js Setup](https://docs.google.com/document/d/1erCdpvykiP3lJaU2jsWvlSO-6UMu6Ty9tiXvu1krQTQ/edit#heading=h.dxq8tktacvg8) documentation for creating migration files.

Knex.js does not support Mongo DB or No SQL Databases. It only support relational databases. 

So I will skip it for now.

**Comment by kimmayorga** [comment]
@iamcubs
Yeah, I just did some research and It supports databases like PostgreSQL, MySQL, SQLite3, and MSSQL, etc., but not Mongo DB

**Comment by kimmayorga** [comment]
@iamcubs further research I think mongo DB has it's own Migration Tool, can you look to this **Migrate-mongo**

**Comment by iamcubs** [comment]
Thank you on this @kimmayorga. I will try to create a migration file using this migrate-mongoose.

**Comment by kimmayorga** [comment]
@iamcubs let me know if I can help on anything.

**Comment by jealapera** [comment]
@iamcubs Thank you for bringing this up, apologies for the oversight. Please skip the creation of migration files and proceed with directly updating the model schema via Mongoose within the code. I've also updated the task description accordingly for your reference.

CC: @kimmayorga

**Comment by iamcubs** [comment]
Noted on this @jealapera . Thank you.
